Jenkinsfile

```diff
@@ -0,0 +1,77 @@
+pipeline {
+  agent {
+    dockerfile {
+      additionalBuildArgs '--build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g)'
+    }
+  }
+  stages {
+    stage("Init title") {
+      when { changeRequest() }
+      steps {
+        script {
+          currentBuild.displayName = "PR ${env.CHANGE_ID}: ${env.CHANGE_TITLE}"
+        }
+      }
+    }
+    stage('Build') {
+      steps {
+        ansiColor('xterm') {
+          sh '''
+            git submodule update --init
```

Comment:
I'd rather this be handled by the Jenkins job config so we consistently run all our vcs outside the container.

Reply:
I would as well, but unfortunately that leads to a recursive submodule structure because of the VSC repository. So the SCM checkout does a normal `--init`, I make sure to do again here, and then manually do a `--recursive` inside .build/k.